solr_plugins/Schlagwortfolgensuche/src/de/uni_tuebingen/ub/ixTheo/keywordChainSearch/KeywordChainParserParser.java

```diff
@@ -7,27 +7,70 @@
 import org.apache.solr.request.SolrQueryRequest;
 import org.apache.solr.search.QParser;
 import org.apache.solr.search.SyntaxError;
+import java.util.ArrayList;
+import java.util.Arrays;
+
 
 public class KeywordChainParserParser extends QParser {
 
     private Query innerQuery;
+    private String origQueryString;
 
     public KeywordChainParserParser(final String searchString, final SolrParams localParams, final SolrParams params, final SolrQueryRequest request) {
         super(searchString, localParams, params, request);
         try {
-	    final String queryString = "key_word_chain_bag:(" + searchString +")";
-	    final QParser parser = getParser(queryString, "lucene", getReq());
-	    this.innerQuery = parser.parse();
-        } catch (SyntaxError ex) {
+        	this.origQueryString = searchString;
+           	final String queryString = "key_word_chain_bag:(" + adjustQueryString(origQueryString) +")";
+	    
+        	//String reqString = getReq().getParamString();
+	    
+        	final QParser parser = getParser(queryString, "lucene", getReq());
+        	this.innerQuery = parser.parse();
+        } 
+        catch (SyntaxError ex) {
             throw new RuntimeException("error parsing query", ex);
         }
     }
 
 
     public Query parse() throws SyntaxError {
-        return this.innerQuery;
+    	// return this.innerQuery;
+    	return new KeywordChainSearchQuery(innerQuery, origQueryString);
     }
+    
+    
+    public String adjustQueryString(String origQueryString){
+    	
+    	// We would like to flexible as to the matches in the key_word_chain_bag to acquire 
+    	// a broad set for the Scorer to compare on, so truncate the terms of the original search
+    	// for the lucene search
+    	
+    	final int PREFIXLEN = 4;
+    	
+    	origQueryString = origQueryString.replaceAll("[^\\p{L}\\p{Nd}\\s]+", "");
+	    ArrayList<String> queryList = new ArrayList<String>(Arrays.asList(origQueryString.split("[\\s]")));
+	    
+	    String newQueryString = "";
```

Comment:
Laut http://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java passiert das bei String concatenation automatisch, falls es dem Compiler sinnvoll erscheint.